### PR TITLE
Cocina-izes Goobi service.

### LINKED ISSUE #3173: Rewrite Dor::Goobi service to use Cocina objects

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3173

## Why was this change made?
Switch Goobi service from using Fedora objects to Cocina objects.

## How was this change tested?
Unit

### CODE REVIEW

File: app/models/dor/goobi.rb [hunk 1]
```diff
@@ -39,14 +41,14 @@ def goobi_xml_tags
     def xml_request
       <<-END
         <stanfordCreationRequest>
-            <objectId>#{@druid_obj.id}</objectId>
-            <objectType>#{@druid_obj.object_type}</objectType>
-            <sourceID>#{@druid_obj.source_id.encode(xml: :text)}</sourceID>
+            <objectId>#{cocina_obj.externalIdentifier}</objectId>
+            <objectType>item</objectType>
```
Comment (justinlittman) [inline app/models/dor/goobi.rb:45]: @andrewjbtw Can this be hardcoded to item?

File: app/services/cocina/to_fedora/descriptive.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def transform
           xml.mods(mods_attributes) do
             Descriptive::DescriptiveWriter.write(xml: xml, descriptive: descriptive, druid: druid)
           end
-        end
+        end.doc
```
Comment (justinlittman) [inline app/services/cocina/to_fedora/descriptive.rb:26]: This was returning a Builder, not a Document (as the method doc specified). Correcting this required removing `.doc` in a handful of locations below.

File: app/models/dor/goobi.rb (hunk 1 above)
Comment (andrewjbtw) [reply app/models/dor/goobi.rb:45]: What does this do? Is this part of the communication with goobi?

Comment (justinlittman) [reply app/models/dor/goobi.rb:45]: We've determined that this can be hardcoded.

**Review by peetucket — APPROVED** [review]

**Comment by peetucket** [comment]
LGTM - not sure if we need to hold for another review or testing on stage before merging

**Pull request merged**